docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,517 @@
+# NodeAgentConfig configuration with restic/kopia
+Date: 2023-07-25
+
+## Abstract
+
+New configuration for the file system backup & restore that will be used by the OADP and allow to choose Restic or Kopia uploader type.
+
+## Background
+
+For the file system backup and restore the Velero may use Restic or Kopia as the uploader mechanism. A number of tests were [performed](https://velero.io/docs/main/performance-guidance/) by the Velero community to compare those mechanisms. In many cases Kopia uploader is a much better performing mechanism and as such should be added to the OADP operator.
+
+Please refer to the upstream [kopia uploader integration design](https://github.com/vmware-tanzu/velero/blob/main/design/unified-repo-and-kopia-integration/unified-repo-and-kopia-integration.md) for underlying Velero design and backup & restore workflow.
+
+Upstream [kopia](https://github.com/kopia/kopia) project which is used by Velero and configured by the OADP as this design proposes.
+
+## Goals
+
+- A new option `nodeagentconfig` to allow configuration of restic or kopia uploader
+- Backwards compatibility with the current OADP configuration schema options, namely `restic`
+- Preparation for the future deprecation of the current `restic` configuration option (from OADP 1.4+)
+- Allow new schema(s) to be used by the datamover node agent
+
+## Non Goals
+- Enablment of datamover node agent
+- Deprecation of the "restic" configuration option
```

Comment:
I think it's meant to say, out of scope.. for now.

Reply:
Sure -- but we are deprecating "restic" config now. As of 1.3, restic config is deprecated and users are encouraged to use nodeAgentConfig instead, so we will print deprecation warnings for use of restic config in 1.3. We want to deprecate it now so 1.3 users will get warnings that it goes away in the future, so that they have more time to move to the new config.

Reply:
So deprecation of restic config is in scope for this proposal.

Reply:
Ok understood. We are deprecating, but we are not removing it. So the goal is to deprecate, non goal is to remove it.

Reply:
moved to goals, non-goal added:
`- Removal of the "restic" configuration option`